pkg/hydrator/batch.go

```diff
@@ -47,6 +47,47 @@ func (b *batchSolutionFetcher) BatchFetchSolution(ctx context.Context, solutionI
 	return solutions, nil
 }
 
+// BatchCheckFetcher represents an interface for concurrently fetching checks
+type BatchCheckFetcher interface {
+	BatchFetchCheck(ctx context.Context, checkIDs []string) ([]bool, error)
+}
+
+type batchCheckFetcher struct {
+	CheckFetcher CheckFetcher
+}
+
+func (b *batchCheckFetcher) BatchFetchCheck(ctx context.Context, checkIDs []string) ([]bool, error) {
```

Comment:
NIT: I'm not sure why the linter didn't require you to add a comment for this exported function, but it would be nice to have a comment here in this context about what the return `[]bool` is. I'm following your code, but with this naming and without going and looking at `FetchCheck()` to see what it returns, I'd expect a `[]check` to be returned instead of the boolean slice.

Reply:
Agreed that this would be useful. The linter doesn't require one because the type, `batchCheckFetcher`, is not exported, so none of its receiver functions will be.

Aaaand this is why you don't use capitalization to determine private/public attribute access. Oh you, Google Golang.

Will fix!